Marlin/src/module/planner.cpp

```diff
@@ -1934,21 +2141,33 @@ void Planner::_buffer_steps(const int32_t (&target)[XYZE]
       // NOTE: Computed without any expensive trig, sin() or acos(), by trig half angle identity of cos(theta).
       if (junction_cos_theta > 0.999999) {
         // For a 0 degree acute junction, just set minimum junction speed.
-        vmax_junction = MINIMUM_PLANNER_SPEED;
+        vmax_junction_sqr = sq(MINIMUM_PLANNER_SPEED);
       }
       else {
-        junction_cos_theta = max(junction_cos_theta, -0.999999); // Check for numerical round-off to avoid divide by zero.
+        NOLESS(junction_cos_theta, -0.999999); // Check for numerical round-off to avoid divide by zero.
         const float sin_theta_d2 = SQRT(0.5 * (1.0 - junction_cos_theta)); // Trig half angle identity. Always positive.
 
         // TODO: Technically, the acceleration used in calculation needs to be limited by the minimum of the
         // two junctions. However, this shouldn't be a significant problem except in extreme circumstances.
-        vmax_junction = SQRT((block->acceleration * JUNCTION_DEVIATION_FACTOR * sin_theta_d2) / (1.0 - sin_theta_d2));
+        vmax_junction_sqr = (JUNCTION_ACCELERATION_FACTOR * JUNCTION_DEVIATION_FACTOR * sin_theta_d2) / (1.0 - sin_theta_d2);
+        if (block->millimeters < 1.0) {
```

Comment:
Is it worth making the 1.0 configurable?

Reply:
Straight line acceleration is taxing on the linear advance, and conserving momentum through the corner helps. Being able to set a high corner acceleration and a low block->acceleration is pretty useful. Conversely, lowering the corner speed could help ringing, even with a high block->acceleration with the bezier jerk control.

Reply:
In fact, if this was available on the menu, the two numbers could be meaningfully calibrated. Since the cornering code is actually following a corner, and the deviation one is a proxy for the give of the hardware at this point, you can create a rectangle with a tiny chamfer on the corners on only one side. The deviation value can be tuned until the amount of force is similar on all corners, and then the acceration could be tuned for an acceptable level of force. If ringing is unacceptable, the acceleration value can be tuned, but the deviation value is specific to the machine. At some point, we may actually follow the deviated path, in which case it wouldn’t need calibration, but is meaningful to adjust.